test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java

```diff
@@ -174,11 +174,81 @@ public void testRandomUniqueNormalUsageAlwayMoreThanOne() {
         assertThat(randomUnique(() -> randomAlphaOfLengthBetween(1, 20), 10), hasSize(greaterThan(0)));
     }
 
+    public void testRandomSubsetOfWithVarargs() {
+        List<Integer> result = ESTestCase.randomSubsetOf(2, 1, 2, 3, 4);
+        assertEquals(2, result.size());
+        assertTrue(Arrays.asList(1, 2, 3, 4).containsAll(result));
```

Comment:
I think we can use randomness to make some more general statements here. This should hold for all values of `2` (the subset size), so why not choose it randomly? Likewise the source list could be of any length, it doesn't have to be `{1,2,3,4}`.